FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs

```diff
@@ -94,7 +95,16 @@ public sealed class FhirApiDataClient : IFhirDataClient
         /// <returns>Uri with search parameters.</returns>
         private Uri CreateSearchUri(FhirSearchParameters searchParameters)
         {
-            var baseUri = new Uri(_dataSource.FhirServerUrl);
+            // If the baseUri has relative parts (like /api),
+            // then the relative part must be terminated with a slash, (like /api/),
+            // if the relative part of baseUri is to be preserved in the constructed Uri. See https://docs.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-6.0
+            var serverUrl = _dataSource.FhirServerUrl;
+            if (!_dataSource.FhirServerUrl.EndsWith("/"))
+            {
+                serverUrl = $"{_dataSource.FhirServerUrl}/";
+            }
+
+            var baseUri = new Uri(serverUrl);
```

Comment:
Do we always combine the search parameters with base url? Will the customer want to search Patient resource like ```https://example.com/a/b/c/Patient```?

Reply:
When customer searches Patient, the final query url will look like `https://example.com/a/b/c/Patient?_lastUpdated=gtxxx&ct=xxxx&_sort=_lastUpdated....`

Reply:
Sorry I just thought the ```baseUri``` here is the host, got it now.